testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -3290,104 +3226,474 @@ fn tx_replay_forking_test() {
+    info!(
+        "---- Waiting for block pushed at height: {:?} ----",
+        expected_height
+    );
+
+    let block = wait_for_block_pushed_by_miner_key(60, expected_height, &stacks_miner_pk)
+        .expect("Timed out waiting for block pushed after fork");
+
+    info!("---- Block: {:?} ----", block);
+
+    for (block_tx, expected_txid) in block
+        .txs
+        .iter()
+        .filter(|tx| {
+            // In this case, the miner issued a tenure extend in the block,
+            // because it's continuing a late tenure.
+            !matches!(
+                tx.payload,
+                TransactionPayload::TenureChange(TenureChangePayload {
+                    cause: TenureChangeCause::Extended,
+                    ..
+                })
+            )
+        })
+        .zip(expected_tx_replay_txids.iter())
+    {
+        assert_eq!(block_tx.txid().to_hex(), *expected_txid);
+    }
+
     signer_test
-        .running_nodes
-        .btc_regtest_controller
-        .build_next_block(3);
+        .wait_for_signer_state_check(30, |state| Ok(state.get_tx_replay_set().is_none()))
+        .expect("Timed out waiting for tx replay set to be cleared");
 
-    let burn_blocks = test_observer::get_burn_blocks();
-    let forked_blocks = burn_blocks.iter().rev().take(2).collect::<Vec<_>>();
-    let last_forked_tenure: ConsensusHash = hex_bytes(
-        &forked_blocks[0]
-            .get("consensus_hash")
-            .unwrap()
-            .as_str()
-            .unwrap()[2..],
-    )
-    .unwrap()
-    .as_slice()
-    .into();
-    let first_forked_tenure: ConsensusHash = hex_bytes(
-        &forked_blocks[1]
-            .get("consensus_hash")
-            .unwrap()
-            .as_str()
-            .unwrap()[2..],
-    )
-    .unwrap()
-    .as_slice()
-    .into();
+    signer_test.shutdown();
+}
 
-    let fork_info = signer_test
-        .stacks_client
-        .get_tenure_forking_info(&first_forked_tenure, &last_forked_tenure)
-        .unwrap();
+#[test]
+#[ignore]
+/// Trigger a Bitcoin fork and ensure that the signer
+/// both detects the fork and moves into a tx replay state
+/// and causes the miner to mine the appropriate list of
+/// transactions in the subsequent blocks
+///
+/// The test flow is:
+///
+/// - Mine 10 tenures after epoch 3
+/// - Include a STX transfer in the 10th tenure
+/// - Trigger a Bitcoin fork (3 blocks)
+/// - Verify that the signer moves into tx replay state
+/// - Verify that the signer correctly includes the stx transfer
+///   in the tx replay set
+/// - Force the miner to ignore replay transactions and attempt
+///   to mine a regular block
+/// - Verify the signers reject this proposed block due to it
+///   missing the replay transactions
+/// - Allow the miner to consider the replay transactions
+/// - Verify the miner correctly constructs a block containing the
+///   tx replay set
+/// - Verify the signers approve subsequent blocks
+fn tx_replay_e2e_test() {
```

Comment:
nit: The name `tx_replay_e2e_test` feels a bit too generic. Could we use a more descriptive name that reflects the specific scenario being tested? 

Could this suggestion better convey the intent of the test?
```suggestion
fn tx_replay_ongoing_rejects_regular_block_proposal() {
```